src/routes/Landing.js

```diff
@@ -41,30 +47,80 @@ class Landing extends Component {
                 this.setState({ unauthed: false });
             } else {
                 this.setState({ unauthed: true });
+                this.setState({ iframeFlag: true });
             }
         }).catch(() => {
             this.setState({ unauthed: true });
+            this.setState({ iframeFlag: false });
         });
+    }
+
+    iframeStyle = {
+        height: '0',
+        width: '0',
+        border: 'none'
+    };
+
+    iframeSrc = <p></p>;
 
+    iFrameCheckAuth = () =>{
+        if (this.state.iframeFlag) {
+            window.insights.chrome.auth.checkAuth();
+        }
+    }
+
+    handleIframeToggle = () => {
+        this.setState({ iframeFlag: false });
```

Comment:
This is not good, because what you are doing here is once the iframe loads you set this flag to false, but at that time user is no prepared yet so it will land on marketing page.

Simplest way how to solve this could be with
```JS
handleIframeToggle = () => {
        window.top.insights.chrome.auth.getUser().then((user) => {
            if (user !== undefined) { // only if getUser return undefined
                this.setState({ iframeFlag: false })
            }
        })
    }
```